Sources/PackageRegistry/SourceArchiver.swift

```diff
@@ -0,0 +1,72 @@
+/*
+ This source file is part of the Swift.org open source project
+
+ Copyright (c) 2021 Apple Inc. and the Swift project authors
+ Licensed under Apache License v2.0 with Runtime Library Exception
+
+ See http://swift.org/LICENSE.txt for license information
+ See http://swift.org/CONTRIBUTORS.txt for Swift project authors
+ */
+
+import Basics
+import TSCBasic
+import TSCUtility
+
+import Dispatch
+
+/// An `Archiver` that handles source archives.
+///
+/// Source archives created with `swift package archive-source`
+/// have a top-level directory prefix (for example, `LinkedList-1.1.1/`).
+/// Unfortunately, the `unzip` command used by `ZipArchiver` doesn't have an option for
+/// ignoring this top-level prefix.
+/// Rather than performing additional (possibly unsafe) file system operations,
+/// this `Archiver` delegates to `tar`, which has a built-in `--strip-components=` option.
+struct SourceArchiver: Archiver {
+    public var supportedExtensions: Set<String> { ["zip"] }
+
+    /// The file-system implementation used for various file-system operations and checks.
+    private let fileSystem: FileSystem
+
+    /// Creates a `SourceArchiver`.
+    ///
+    /// - Parameters:
+    ///   - fileSystem: The file-system to used by the `SourceArchiver`.
+    public init(fileSystem: FileSystem = localFileSystem) {
```

Comment:
trying to avoid defaulting to localFileSystem where we can

Reply:
Fixed by b1a641f5